File: human_readable_time/elixir/human_readable_time.exs
defmodule HumanReadable do
  @sec_in_a_min 60
  @sec_in_a_hour 3600

  @spec format(seconds: non_neg_integer()) :: Time.t()
  def format(seconds) when seconds <= 359999 do
    sec =
      rem(seconds, @sec_in_a_hour)
      |> rem(@sec_in_a_min)
      |> Integer.to_string()
      |> format_zero()

    min =
      rem(seconds, @sec_in_a_hour)
      |> div(@sec_in_a_min)
      |> Integer.to_string()
      |> format_zero()

    hours =
      div(seconds, @sec_in_a_hour)
      |> Integer.to_string()
      |> format_zero()

    "#{hours}:#{min}:#{sec}"
  end

  defp format_zero(str), do: if String.length(str) < 2, do: "0#{str}", else: str
end

IO.inspect(HumanReadable.format(3600))

File: human_readable_time/rust/human_readable_time.rs
fn make_readable(seconds: u32) -> String {
    const SEC_IN_MIN: u32 = 60;
    const SEC_IN_HOUR: u32 = 3600;

    let mut secs: String = ((seconds % SEC_IN_HOUR) % SEC_IN_MIN).to_string();

    if secs.chars().count() < 2 {
        secs = format!("0{}", secs)
    }

    let mut min: String = ((seconds % SEC_IN_HOUR) / SEC_IN_MIN).to_string();

    if min.chars().count() < 2 {
        min = format!("0{}", min)
    }

    let mut hours: String = (seconds / SEC_IN_HOUR).to_string();

    if hours.chars().count() < 2 {
        hours = format!("0{}", hours)
    }

    format!("{}:{}:{}", hours, min, secs)
}

fn main() {
    println!("{:?}", make_readable(3600))
}